mantidproject/mantid
# Engg:: Stop calibration if peaks cannot be found

Description of work.
This pull request makes the Engineering Diffraction interface stop calibrating if `EnggFitPeaks` fails to find any peaks. As the algorithm was not set to re-throw the try catch block was not doing anything so instead we check if it successfully executed. 

**To test:**
Open the `Engineering Diffraction` interface
The input files are located at `\\olympic\Babylon5\Public\David Fairbrother\z_test`

Enter Vanadium run: `256355`
Enter Cerium run: `255415`
Run calibrate. This should fail and not produce an output. 

---

There are two additional minor fixes which were completed with this issue.
The `Engineering Diffraction Testing Guide` should now appear under the interfaces category in documentation.

On opening and closing the `Engineering Interface` in Mantid the temporary workspace `engggui_calibration_banks_parameters` should now be removed.

Fixes #17282 and #17271 .

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
